model/task_history.go

```diff
@@ -233,3 +235,17 @@ func getOlderActiveMainlineTask(ctx context.Context, t task.Task) (*task.Task, e
 	// olderActiveTask can be nil, since it may not exist.
 	return olderActiveTask, nil
 }
+
+func GetTaskDateOrder(ctx context.Context, date time.Time, opts FindTaskHistoryOptions) (int, error) {
```

Comment:
Can we rename this (and also GetVersionDateOrder) to something like GetRevisionOrderByDate() or something like that? That will help remove ambiguity since order can also imply tasks ordered in a certain way.